js/validators.js

```diff
@@ -0,0 +1,70 @@
+const validator = {
+    email: validateEmail,
+    nickname: validateEmpty,
+    password: validatePassword,
+    passwordrepeat: validatePasswordRepeat,
+};
+
+export function validateInput(value, fieldName) {
+    let validateFunction = validator[fieldName];
+    return validateFunction(value, fieldName);
+}
+
+function isEmpty(value) {
+    return value.trim() === '';
+}
+
+function isValidEmail(value) {
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+
+    return emailRegex.test(value);
+}
+
+function validateEmpty(value, fieldName) {
+    let fieldNameText = '';
+    if (fieldName === 'password' || fieldName === 'passwordrepeat') {
+        fieldNameText = '비밀번호';
+
+        return isEmpty(value) ? `${fieldNameText}를 입력해주세요.` : ``;
+    } else if (fieldName === 'nickname') {
+        fieldNameText = '닉네임';
+    } else if (fieldName === 'email') {
+        fieldNameText = '이메일';
+    }
+
+    return isEmpty(value) ? `${fieldNameText}을 입력해주세요.` : ``;
+}
```

Comment:
여러 `input` 컴포넌트에서 쓰이는 공통 로직을 하나의 함수로 잘 정리해주셨습니다. 🤗
다만, `~를 입력해 주세요 부분`이 중복되어 쓰이고 있고, default case에 대한 처리가 없는 것이 아쉽습니다.

```suggestion
function getFieldNameText(fieldName) {
    switch (fieldName) {
        case 'password':
        case 'passwordrepeat':
            return '비밀번호';
        case 'nickname':
            return '닉네임';
        case 'email':
            return '이메일';
        default:
            return '필드';
    }
}

function validateEmpty(value, fieldName) {
    const fieldNameText = getFieldNameText(fieldName);
    return isEmpty(value) ? `${fieldNameText}을(를) 입력해주세요.` : '';
}
```
이런식으로 필드별 안내메세지에 들어갈 단어를 분기처리하는 부분과 안내메세지 부분을 별도의 함수로 분리해보셔도 좋을 것 같아요!